crypto/s2n_composite_cipher_aes_sha.c

```diff
@@ -26,24 +26,29 @@
 #include "utils/s2n_safety.h"
 #include "utils/s2n_blob.h"
 
-/* LibreSSL, BoringSSL and AWS-LC support the cipher, but the interface is different from Openssl's. We
- * should define a separate s2n_cipher struct for LibreSSL, BoringSSL and AWS-LC.
+/* LibreSSL and BoringSSL support the cipher, but the interface is different from Openssl's. We
+ * should define a separate s2n_cipher struct for LibreSSL and BoringSSL.
  */
-#if !defined(LIBRESSL_VERSION_NUMBER) && !defined(OPENSSL_IS_BORINGSSL) && !defined(OPENSSL_IS_AWSLC)
+#if !defined(LIBRESSL_VERSION_NUMBER) && !defined(OPENSSL_IS_BORINGSSL)
 /* Symbols for AES-SHA1-CBC composite ciphers were added in Openssl 1.0.1
  * These composite ciphers exhibit erratic behavior in LibreSSL releases.
  */
-#if S2N_OPENSSL_VERSION_AT_LEAST(1,0,1) 
+#if S2N_OPENSSL_VERSION_AT_LEAST(1,0,1)
```

Comment:
Is this true for AWS-LC when in fips mode because of https://github.com/awslabs/aws-lc/pull/520? (#520 is correct, but I think we also need something here as well)

Reply:
In FIPS mode, the API returns NULL. s2n side has runtime check. This runtime check is needed in other cases(e.g. if the hardware supports the impl).
```
const EVP_CIPHER *EVP_aes_128_cbc_hmac_sha256(void) {
  return hwaes_capable() ? &aesni_128_cbc_hmac_sha256_cipher : NULL;
}

const EVP_CIPHER *EVP_aes_256_cbc_hmac_sha256(void) {
  return hwaes_capable() ? &aesni_256_cbc_hmac_sha256_cipher : NULL;
}
```